core/src/main/scala/kafka/utils/LoggingController.scala

```diff
@@ -112,24 +112,30 @@ private class Log4jCoreController extends LoggingControllerDelegate {
     val level = Level.toLevel(logLevel.toUpperCase(Locale.ROOT))
 
     if (loggerName == ROOT_LOGGER) {
-      Configurator.setAllLevels(LogManager.ROOT_LOGGER_NAME, level)
+      Configurator.setLevel(LogManager.ROOT_LOGGER_NAME, level)
       true
     } else {
       if (loggerExists(loggerName) && level != null) {
-        Configurator.setAllLevels(loggerName, level)
+        Configurator.setLevel(loggerName, level)
         true
       }
       else false
     }
   }
 
   override def unsetLogLevel(loggerName: String): Boolean = {
+    val nullLevel: String = null
     if (loggerName == ROOT_LOGGER) {
-      Configurator.setAllLevels(LogManager.ROOT_LOGGER_NAME, null)
+      Configurator.setLevel(LogManager.ROOT_LOGGER_NAME, nullLevel)
       true
     } else {
       if (loggerExists(loggerName)) {
-        Configurator.setAllLevels(loggerName, null)
+        val parent = logContext.getLogger(loggerName).getParent
+        if (parent == null) {
+          Configurator.setLevel(loggerName, nullLevel)
+        } else {
+          Configurator.setLevel(loggerName, parent.getLevel)
+        }
```

Comment:
I try to align with the behavior of the old version of Kafka here.
https://github.com/apache/kafka/blob/66e2ac0e098f6ee83deaf5501bf3f606d29b35fa/core/src/main/scala/kafka/utils/Log4jController.scala#L33-L49